README.md

```diff
@@ -95,56 +87,20 @@ let device: Device = instance
     .unwrap();
 ```
 
-To not lose this lifetime single items can be "cast" to a slice of length _one_ with `std::slice::from_ref` while still taking advantage of `Deref`:
-
-```rust
-let queue_info = vk::DeviceQueueCreateInfo::builder()
-    .queue_family_index(queue_family_index)
-    .queue_priorities(&priorities);
-
-let device_create_info = vk::DeviceCreateInfo::builder()
-    .queue_create_infos(std::slice::from_ref(&queue_info))
-    ...;
-```
-
-Builders have an explicit lifetime, and are marked as `#[repr(transparent)]`.
-```rust
-#[repr(transparent)]
-pub struct DeviceCreateInfoBuilder<'a> {
-    inner: DeviceCreateInfo,
-    marker: ::std::marker::PhantomData<&'a ()>,
-}
-impl<'a> DeviceCreateInfoBuilder<'a> {
-    //...
-    pub fn queue_create_infos(
-        mut self,
-        queue_create_infos: &'a [DeviceQueueCreateInfo],
-    ) -> DeviceCreateInfoBuilder<'a> {...}
-    //...
-```
-
-Every reference has to live as long as the builder itself. Builders implement `Deref` targeting their corresponding Vulkan struct, so references to builders can be passed directly
-to Vulkan functions.
-
-Calling `.build()` will **discard** that lifetime because Vulkan structs use raw pointers internally. This should be avoided as much as possible because this can easily lead to dangling pointers. If `.build()` has to be called, it should be called as late as possible. [Lifetimes of temporaries](https://doc.rust-lang.org/reference/expressions.html#temporary-lifetimes) are extended to the enclosing statement, ensuring they are valid for the duration of a Vulkan call occurring in the same statement.
-
-
 ### Pointer chains
 
 ```rust
-let mut variable_pointers = vk::PhysicalDeviceVariablePointerFeatures::builder();
-let mut corner =
-    vk::PhysicalDeviceCornerSampledImageFeaturesNV::builder();
+let mut variable_pointers = vk::PhysicalDeviceVariablePointerFeatures::default();
+let mut corner = vk::PhysicalDeviceCornerSampledImageFeaturesNV::default();
 ;
-let mut device_create_info = vk::DeviceCreateInfo::builder()
+let mut device_create_info = vk::DeviceCreateInfo::default()
     .push_next(&mut corner)
     .push_next(&mut variable_pointers);
 ```
 
-Pointer chains in builders differ from raw Vulkan. Instead of chaining every struct manually, you instead use `.push_next` on the struct that you are going to pass into the function. Those structs then get *prepended* into the chain.
-
-`push_next` is also type checked, you can only add valid structs to the chain. Both the structs and the builders can be passed into `push_next`. Only builders for structs that can be passed into functions will implement a `push_next`.
+Pointer chains in Ash differ from raw Vulkan. Instead of chaining every struct manually, you instead use `.push_next()` on the struct that you are going to pass into the function. Those structs then get _prepended_ into the chain.
```

Comment:
Nit: I don't think "Pointer chains in Ash differ from raw Vulkan" is the right wording, but it was already there. They are still the same pointer chains, it's just that our `.push_next()` helper function has a specific way to "`.insert(1)`" rather than `.append()` to be simple and efficient (no need to walk to the end of the chain, though we do need to do that for the struct that is being appended, but the common case is that it's just one item).

At the same time [the spec says](https://registry.khronos.org/vulkan/specs/1.3/styleguide.html#extensions-interactions) that specific `pNext` ordering is allowed (but strongly discouraged) if drivers/extensions absolutely need it for deterministic behaviour.

Reply:
I agree this wording could be improved, both for clarity and concision. Perhaps:

> Ash provides helpers for assembling pointer chains. Use `.push_next()` on the struct ...

Reply:
Updated and reordered. Left out the "refer to the `push_next()` documentation for more detailed insertion order" or something along those lines.

Note that our `fn push_next()` docs don't cover the case where a hypothetical `D->E` chain is appended to an `A->B->C` chain, which will end up looking like `A->D->E->B->C`.